src/Thorn_Calculus-Core_Bisimilarities.thy

```diff
@@ -1350,9 +1350,148 @@ text \<open>
   commutativity rules.
 *)
 
+lemma repeated_receive_evolution:
+  assumes "A \<triangleright>\<^sup>\<infinity> x. \<P> x \<rightarrow>\<^sub>s\<lparr>\<alpha>\<rparr> Q"
+  obtains n and X
+  where "\<alpha> = A \<triangleright> \<star>\<^bsup>n\<^esup> X"
+    and "Q = post_receive n X (\<lambda>v. \<P> v \<parallel> A \<triangleright>\<^sup>\<infinity> x. \<P> x)"
```

Comment:
We should avoid manual alignment like that of aligning the term after `where` with the term after `and`, because it almost certainly breaks in PDF output. In this particular case, I would just put the code in a single line.

Reply:
Also, the use of the variable name `v` isn’t in line with the, still unwritten, conventions that I followed in my code. Letters `v` and `w` are for completely generic things; in the `Thorn_Calculus-Foundations` theory, they are used for things that can be channels, values, and processes in particular. The names used for values are `x` and `y`. Since what you’ve named `v` above is what’s named `x` in the assumption and problems with shadowing cannot occur, I’d just replace `v` by `x`.

Reply:
Fixed in 7a7c10bec73f3fbbae1b23ee24e9534bb5fe0531 and 2ef937d043006776d361fe000efc44a64c975eaa, please verify.

Reply:
7a7c10bec73f3fbbae1b23ee24e9534bb5fe0531 looks good to me. 2ef937d043006776d361fe000efc44a64c975eaa is kind of incomplete, since there are other occurrences of `v` that should be replaced by `x`.

Reply:
Oh, I see that the further replacements of `v` by `x` are handled in 47bd854e92b49c095c0e82208156126a1001e56a. So this all looks good to me now.